src/oop/programs_with_multiple_classes/medium/q5/MainTest.java

```diff
@@ -0,0 +1,122 @@
+        assertTrue(desk.hasField(varMaterial, String.class), modifiedDeskMessage);
+        assertTrue(desk.hasModifier(varMaterial, "private"), modifiedDeskMessage);
+        assertTrue(desk.hasField(varMonitors, Monitor[].class), modifiedDeskMessage);
+        assertTrue(desk.hasModifier(varMonitors, "private"), modifiedDeskMessage);
+
+        String modifiedMonitorMessage =
+                "You have modified the provided portions of class " + monitorClass + ". Please revert them to the original state.";
+        assertTrue(monitor.hasField(varCount, int.class), modifiedMonitorMessage);
+        assertTrue(monitor.hasModifier(varCount, "private"), modifiedMonitorMessage);
+        assertTrue(monitor.hasField(varOn, boolean.class), modifiedMonitorMessage);
+        assertTrue(monitor.hasModifier(varOn, "private"), modifiedMonitorMessage);
+        Object[][] tests = new Object[][]{
+                {343, true},
+                {3434, false},
+                {7653, true}
+        };
+        Class<?>[] classArguments = {int.class, boolean.class};
+        assertTrue(monitor.hasConstructor(classArguments), modifiedMonitorMessage);
+        assertTrue(monitor.hasModifier(classArguments, "public"), modifiedMonitorMessage);
+        for (int i = 0; i < tests.length; i++) {
+            Object[][] arguments = {
+                    {tests[i][0], int.class},
+                    {tests[i][1], boolean.class}
+            };
+            Object chairInstance = monitor.createInstance(arguments);
+            _assertEquals(tests[i][0], monitor.getFieldValue(chairInstance, varCount),
+                    "Your " + monitorClass + " constructor does not correctly initialize the " + varCount + " field.");
+            _assertEquals(tests[i][1], monitor.getFieldValue(chairInstance, varOn),
+                    "Your " + monitorClass + " constructor does not correctly initialize the " + varOn + " field.");
+        }
+    }
+
+    @Test
+    public void deskClassHasRequiredConstructor() {
+        Class<?>[] classArguments = {String.class};
+        assertTrue(desk.hasConstructor(classArguments),
+                "Your " + deskClass + " constructor does not have the correct parameters.");
+        assertTrue(desk.hasModifier(classArguments, "public"),
+                "Your " + deskClass + " constructor does not have the correct modifier.");
+    }
+
+    private static Stream<Arguments> deskInputProvider() {
+        return Stream.of(
+                Arguments.of("Mialani"),
+                Arguments.of("Oak"),
+                Arguments.of("Steel")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("deskInputProvider")
+    public void deskConstructorInitializesValuesCorrectly(String material) throws Throwable {
+        Object[][] arguments = {
+                {material, String.class},
+        };
+        Object chairInstance = desk.createInstance(arguments);
```

Comment:
You should also check the visibility modifier of the constructor

Reply:
We check that in deskClassHasRequiredConstructor()